shared/lib_time.cpp

```diff
@@ -81,10 +80,13 @@ void single_year_to_lifetime_interpolated(
 		}
 	}
 	// In the case of no single year vector, create lifetime vector of zeros
-	else {
-		for (size_t i = 0; i < n_rec_lifetime; i++) {
-			lifetime_from_singleyear_vector.push_back(0);
-		}
+	else if (singleyear_vector.empty() ) {
+	    for (size_t i = 0; i < n_rec_lifetime; i++)
+		    lifetime_from_singleyear_vector.emplace_back(0);
+	}
+	else if (singleyear_vector.size() == 1) {
+	    for (size_t i = 0; i < n_rec_lifetime; i++)
+	        lifetime_from_singleyear_vector.emplace_back(singleyear_vector[0]);
 	}
 }
 
```

Comment:
Please be careful merging into develop as there are several additional additions and modifications to lib_time

Reply:
There were a couple of bug fixes to `single_year_to_lifetime_interpolated`:

1. dt_hour as 8760 * n_years / n_rec_lifetime
2. singleyear_vector.size() can be 0 (though it hasn't in practice, otherwise we'd have triggered the divide by zero error in line 43...), 1, or greater than 1.

Reply:
fixed divide by zero error and use push_back instead of emplace_back since vector operator[] returns a reference (though with fundamental types I don't think it makes a difference but going to push_back for simplicity) in 3bd7767d806b8513836bcdfb8681bc1a6c6895cb

added test for a size=1 case in 9d42af75d70f8d67302f8d8da65fc92fa89dc260